### PR TITLE
Adding pie generator

### PR DESCRIPTION
**Pull request by devnook**
Adding ability to generate svg pies on the server (piegen/pie-svg), and on client (/piegen/pie).

Also changing the build process a bit for the ease of development.

### CODE REVIEW

File: src/server/js/routes/piegen.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ *  limitations under the License
+ *
+ */
+
+const express = require('express');
+const fs = require('fs');
+const Pie = require('../models/pie.js');
+
+const router = new express.Router();
+const svgPath = __dirname + '/../../../shared/views/pie.svg';
```
Comment (sgomes) [inline src/server/js/routes/piegen.js]: Suggest changing this to `` `${__dirname}/../../../shared/views/pie.svg` `` for readability.

File: src/server/js/routes/piegen.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    scripts: ['/js/pie.js'],
+  });
+});
+
+router.get('/pie-svg', (req, res, next) => {
+  fs.readFile(svgPath, (err, result) => {
+    const pie = new Pie();
+    pie.template = result.toString();
+    pie.topping = req.query.topping;
+    res.writeHead(200, {"Content-Type": "image/svg+xml"});
```
Comment (sgomes) [inline src/server/js/routes/piegen.js]: Use `'` instead of `"` in JS (from Google styleguide).

File: package.json [hunk 3]
```diff
@@ -4,12 +4,14 @@
...
   "dependencies": {
     "body-parser": "~1.17.1",
+    "color": "^2.0.0",
     "cookie-parser": "~1.4.3",
     "debug": "~2.6.3",
     "eslint": "^4.4.1",
     "eslint-config-google": "^0.9.1",
     "express": "~4.15.2",
     "hbs": "~4.0.1",
+    "jsdom": "^11.2.0",
```
Comment (sgomes) [inline package.json]: As per `pie.js`, can we remove this dependency?

File: gulpfile.js [hunk 4]
```diff
@@ -34,9 +34,12 @@ const distDir = './dist';
 gulp.task('clean', () => del([distDir]));
 
 gulp.task('eslint', () => gulp.src(`${srcDir}/**/*.js`)
-  .pipe(eslint())
+  .pipe(eslint({
```
[the remark this answers is not in the document]
Comment (devnook) [reply gulpfile.js]: Ah, cool! I needed this to allow backticks for template literals. But I can do it in the yaml file too.

File: src/server/js/models/pie.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ *
+ */
+
+const color = require('color');
+const jsdom = require('jsdom');
+const ingredients = require('../../../../data/ingredients.json');
+
+const {JSDOM} = jsdom;
```
[the remark this answers is not in the document]
Comment (devnook) [reply src/server/js/models/pie.js]: Hm, ok, but this means you get less reusability of the code (e.g. you need 2 svg generators - one with templating on server and one with DOM on client).

[the remark this answers is not in the document]
Comment (sgomes) [reply src/server/js/models/pie.js]: Hmm, interesting, that's a very good point! It sounds like a challenge we might run into in several places while implementing the app.

I would suggest holding off on that change for now, then, and keeping the code as-is. Perhaps we should set up a weekly sync where we discuss interesting issues like this one?

**Comment by devnook** [comment]
PTAL

**Pull request merged**